### PR TITLE
Differentiate donuts

### LINKED ISSUE #43: Donut & Breach Differentiation

### PR DESCRIPTION
**Pull request by wenjiazhang**
### Summary 
Donuts now have different colors. Breaches have assets that match their corresponding donut.
Close #43 

### Testing 
Image from iOS

### Review Focus 

- Reasonable asset handling
- Refactors are valid

### CODE REVIEW

File: source/GameMode.cpp [hunk 1]
```diff
@@ -76,7 +79,9 @@ bool GameMode::init(const std::shared_ptr<cugl::AssetManager>& assets) {
 	playerId = net.getPlayerID();
 	gm.setPlayerId(playerId);
 	// gm.setDonuts(shipModel);
-	donutModel = donuts.at(playerId);
+	donutModel = donuts.at(static_cast<unsigned long>(playerId));
+	donutModel->setColorId(0);
```
Comment (ssorenson130) [inline source/GameMode.cpp]: Maybe I'm missing something here, but why is this hard coded to 0?

Comment (wenjiazhang) [reply source/GameMode.cpp]: Right now the player is always hardcoded to the first color in the JSON file. This is to ensure that the player has an unique color and everyone else is assigned one of the remaining colors. I set it to zero here to ensure that field is at least a sensible number. I should mark it as a TODO to improve later, unless you have suggestions?

Comment (ssorenson130) [reply source/GameMode.cpp]: I don't mind it for now as long as the players' colors are consistent across devices. Is this the case?

File: source/BreachModel.h [hunk 2]
```diff
@@ -14,8 +14,12 @@ class BreachModel {
 	int player;
 	/** Reference to image in SceneGraph for animation */
 	std::shared_ptr<cugl::PolygonNode> sprite;
+	/** Set to true if sprite needs to be updated */
+	bool needSpriteUpdate;
 
    public:
+	/** Default Max Health of a Breach*/
+	const unsigned int HEALTH_DEFAULT = 3;
```
Comment (mt-xing) [inline source/BreachModel.h]: nit: Should probably be a `constexpr` instead of `const`.

File: source/DonutNode.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ void DonutNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch, const Mat4
 	float vel = donutModel->getVelocity() > 2 ? donutModel->getVelocity() - 364.08
 											  : donutModel->getVelocity();
 	if (!donutModel->getUpdated()) {
-		CULog("accelerate %f", vel / 0.9f - donutModel->getLastVel());
+		// CULog("accelerate %f", vel / 0.9f - donutModel->getLastVel());
```
Comment (mt-xing) [inline source/DonutNode.cpp]: nit: Just delete the line

File: source/GameGraphRoot.cpp [hunk 4]
```diff
@@ -25,7 +25,7 @@ constexpr float PI_180 = (float)(M_PI / 180);
 constexpr float BREACH_SCALE = 0.25;
 
 /** The scale of the donut textures. */
-constexpr float DONUT_SCALE = 0.5;
+constexpr float DONUT_SCALE = 0.32;
```
Comment (mt-xing) [inline source/GameGraphRoot.cpp]: Shouldn't there be an `f` at the end of the float, ie `0.32f`?

File: source/GameMode.cpp (hunk 1 above)
Comment (wenjiazhang) [reply source/GameMode.cpp]: Player's color depends on the playerId

File: source/DonutNode.cpp (hunk 3 above)
Comment (wenjiazhang) [reply source/DonutNode.cpp]: I think Sam is fixing this in his branch bc it's his log. But I'll delete it here too.

File: source/GameMode.cpp (hunk 1 above)
Comment (mt-xing) [reply source/GameMode.cpp]: Wait, I'm confused. So is the player's color their player ID, or is it 0?

File: source/GameGraphRoot.cpp (hunk 4 above)
Comment (wenjiazhang) [reply source/GameGraphRoot.cpp]: Fixed

Comment (mt-xing) [reply source/GameGraphRoot.cpp]: Wait, it's not tho...

File: source/BreachModel.h (hunk 2 above)
Comment (wenjiazhang) [reply source/BreachModel.h]: `Constexpr` needs to be instantiated at compile time, `const` can be at run-time. Any reason we're using `constexpr`?

Comment (mt-xing) [reply source/BreachModel.h]: The fact that it's a compile-time constant saves a lookup at runtime. It's a replacement for `#define` macros.

File: source/GameMode.cpp (hunk 1 above)
Comment (wenjiazhang) [reply source/GameMode.cpp]: Player's color depends on their playerId but is not equal to it. ColorId is the idx into the playerColor vector. Device client has "yellow" hardcoded into JSON file, so its colorId is 0(yellow). Everyone else gets a number that is NOT 0 calculated using % with their playerId.

Comment (mt-xing) [reply source/GameMode.cpp]: But isn't the whole point of the game to make each player's color consistent across devices? If I roll past a yellow breach and I'm not yellow, I need to be able to tell the person who is that they need to come fix it. How can we do that if everyone is hard-coded to yellow on their own screen?

Comment (wenjiazhang) [reply source/GameMode.cpp]: Good point lol

**Pull request merged**